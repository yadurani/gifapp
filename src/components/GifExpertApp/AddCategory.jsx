import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ handleAdd }) => {
  const [inputValue, setInputValue] = useState('')
  
  const handleChange = (e) => setInputValue(e.target.value)
  const handleSubmit = (e) => {
    e.preventDefault()
    if(inputValue.trim().length > 2) {
      handleAdd(inputValue)
      setInputValue('')
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        onChange={handleChange}
        placeholder="Search"
        value={inputValue}
        type="text"
        />
    </form>
  )
};

AddCategory.propTypes = {
  handleAdd: PropTypes.func.isRequired
}

export default AddCategory
