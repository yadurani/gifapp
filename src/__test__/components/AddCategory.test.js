import React from 'react';

import { shallow } from 'enzyme';

import AddCategory from '../../components/GifExpertApp/AddCategory';

describe('Probar el componente <AddCategory />', () => {
  const handleAdd = jest.fn()
  let wrapper = shallow(<AddCategory handleAdd={handleAdd}/>);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory handleAdd={handleAdd}/>);
  })

  it('Debe contener el snapchot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('Debe de cambiar la caja de texto', () => {
    const value = 'Hola Mundo'
    wrapper.find('input').simulate('change', { target: { value }});
    expect(wrapper.find('input').prop('value')).toBe(value)
  })

  it('No debe de postear la informacion con submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(handleAdd).not.toHaveBeenCalled();
  })

  it('Debe de llamar el handleAdd, y limpiar la caja de texto', () => {
    const value = 'Hola Mundo'

    wrapper.find('input').simulate('change', { target: { value } });

    wrapper.find('form').simulate('submit', { preventDefault(){} });

    expect(handleAdd).toHaveBeenCalled();
    expect(handleAdd).toHaveBeenCalledTimes(1);
    expect(handleAdd).toHaveBeenCalledWith( expect.any(String) );
    // expect(handleAdd).toHaveBeenCalledWith( expect.any(Function) );
    
    expect(wrapper.find('input').prop('value')).toBe('')

  })
});