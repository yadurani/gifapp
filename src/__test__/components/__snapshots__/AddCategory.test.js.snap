// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar el componente <AddCategory /> Debe contener el snapchot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
